Bonsai.Configuration.Tests/ScriptExtensionsTests.cs

```diff
@@ -0,0 +1,211 @@
+              </Packages>
+              <AssemblyReferences>
+              </AssemblyReferences>
+              <AssemblyLocations>
+                <AssemblyLocation assemblyName="{DummyPackageA.Id}" processorArchitecture="MSIL" location="Packages/{DummyPackageA.Id}.{DummyPackageA.Version}/{DummyPackageA.Id}.dll" />
+                <AssemblyLocation assemblyName="{DummyPackageB.Id}" processorArchitecture="MSIL" location="Packages/{DummyPackageB.Id}.{DummyPackageB.Version}/{DummyPackageB.Id}.dll" />
+              </AssemblyLocations>
+              <LibraryFolders>
+              </LibraryFolders>
+            </PackageConfiguration>
+            """;
+
+        var serializer = new XmlSerializer(typeof(PackageConfiguration));
+        using XmlReader reader = XmlReader.Create(new StringReader(dummyBonsaiConfig));
+        var configuration = (PackageConfiguration)serializer.Deserialize(reader);
+        configuration.ConfigurationFile = Path.GetFullPath("DummyBonsai.config");
+        return configuration;
+    }
+
+    [TestMethod]
+    public void HasDefaultProjectMetadata()
+    {
+        EnsureCleanWorkspace();
+
+        var packageConfiguration = CreateDummyPackageConfiguration();
+        using var extensions = new ScriptExtensions(packageConfiguration, $"{nameof(HasDefaultProjectMetadata)}Output");
+
+        Assert.IsFalse(File.Exists("Extensions.csproj"));
+        ScriptExtensionsProjectMetadata metadata = extensions.LoadProjectMetadata();
+
+        Assert.IsFalse(metadata.Exists);
+        Assert.AreEqual(default(ScriptExtensionsProjectMetadata).GetProjectXml(), metadata.GetProjectXml());
+        Assert.IsFalse(File.Exists("Extensions.csproj")); // Project file should not be written automatically
+    }
+
+    [TestMethod]
+    public void WillLoadProjectMetadata()
+    {
+        EnsureCleanWorkspace();
+
+        File.WriteAllText
+        (
+            "Extensions.csproj",
+            """
+            <Project Sdk="Microsoft.NET.Sdk">
+              <PropertyGroup>
+                <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+                <TargetFramework>net472</TargetFramework>
+              </PropertyGroup>
+
+              <ItemGroup>
+                <PackageReference Include="FakePackage" Version="1.0.0" />
+              </ItemGroup>
+            </Project>
+            """
+        );
+
+        var packageConfiguration = CreateDummyPackageConfiguration();
+        using var extensions = new ScriptExtensions(packageConfiguration, $"{nameof(WillLoadProjectMetadata)}Output");
+        ScriptExtensionsProjectMetadata metadata = extensions.LoadProjectMetadata();
+        Assert.IsTrue(metadata.Exists);
+        Assert.IsTrue(metadata.AllowUnsafeBlocks);
+        CollectionAssert.Contains(metadata.GetPackageReferences().ToArray(), "FakePackage");
+    }
+
+    [TestMethod]
+    public void UpdateProjectMetdataInitializesProject()
```

Comment:
```suggestion
    public void UpdateProjectMetadataInitializesProject()
```

There were actually two minor typos in this method name and only one of them was fixed, sorry I didn't make that clear in the first suggestion.

Reply:
Fixed! My bad for not double-checking 😅